daemon/settings.c

```diff
@@ -507,6 +525,29 @@ settings_userdata_path(uid_t uid)
                            (unsigned)uid);
 }
 
+static void
+settings_remove_stale_userdata(uid_t uid)
+{
+    gchar *path = settings_userdata_path(uid);
+    if( g_file_test(path, G_FILE_TEST_EXISTS) ) {
+        if( unlink(path) == -1 )
+            log_err("%s: could not remove: %m", path);
```

Comment:
I'd suggest skipping the test and doing unconditional unlink + complain on errno != ENOENT as it is a bit, a bit faster and avoids (a bit theoretical) toc-tou hazards too.

Reply:
Changed this.